### PR TITLE
Add Refaster template recipe docs

### PR DESCRIPTION
**Pull request by mike-solomon**
Fixes: https://github.com/openrewrite/rewrite-docs/issues/256

### CODE REVIEW

File: authoring-recipes/refaster-recipes.md [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+* There are multiple methods with the same return type
+* All arguments in those methods are identical and have the same name
+* One of the methods is annotated with `@AfterTemplate`
+* Every other method is annotated with `@BeforeTemplate`
+
+Here is an example of what a Refaster template looks like:
+
+```java
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.AlsoNegation;
```
[the remark this answers is not in the document]
Comment (timtebeek) [reply authoring-recipes/refaster-recipes.md:22]: We don't support `@AlsoNegation` just yet
- https://github.com/openrewrite/rewrite-templating/issues/47
```suggestion
```

File: authoring-recipes/refaster-recipes.md [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+    return string.equals("");
+  }
+
+  @BeforeTemplate
+  boolean lengthEquals0(String string) {
+    return string.length() == 0;
+  }
+
+  @AfterTemplate
+  @AlsoNegation
```
Comment (timtebeek) [inline authoring-recipes/refaster-recipes.md:37]:
```suggestion
```

File: authoring-recipes/refaster-recipes.md [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+
+## How to create a Refaster recipe
+
+{% hint style="info" %}
+Moderne provides a [starter recipe repository](https://github.com/moderneinc/rewrite-recipe-starter) which contains all of the below code. We'd recommend using that template repository as a getting started point.
+{% endhint %}
+
+### Update your dependencies
+
+The first thing you'll need to do is update your dependencies to. Below are the minimum recommended dependencies to include in your project:
```
Comment (timtebeek) [inline authoring-recipes/refaster-recipes.md:60]:
```suggestion
The first thing you'll need to do is update your dependencies  and add an annotation processor. Below are the minimum recommended dependencies to include in your project:
```

File: authoring-recipes/refaster-recipes.md [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+                  boolean booleanExpression() {
+                    return true;
+                  }
+              }
+              """
+          )
+        );
+    }
+}
+```
```
Comment (timtebeek) [inline authoring-recipes/refaster-recipes.md:395]: This a fairly sudden end; perhaps we can add a small section to say that this is a great way to get started with recipe development, but if you need more you can write a custom visitor, taking the generated recipe as a base to work from.

**Review by timtebeek — APPROVED** [review]
Great work! Some small suggestions but otherwise good to merge as you see fit.

File: authoring-recipes/refaster-recipes.md (hunk 4 above)
Comment (mike-solomon) [reply authoring-recipes/refaster-recipes.md:393]: Makes sense - will merge in and then add so I can get around git stuff locally 😆 - thanks for the review

**Pull request merged**